src/Lucene.Net/Search/FieldCacheImpl.cs

```diff
@@ -73,11 +74,17 @@ private void  Init()
             _stringIndexCache = new StringIndexCache(this);
         }
 
-        public virtual void  PurgeAllCaches()
+        public virtual IDisposable PurgeAllCaches()
         {
             lock (this)
             {
+                // PurgeAllCaches is replacing the cache with a new one (without actually releasing any memory).
+                // When the GC will run, the finalizer of the Segments will be executed and release the unmanaged memory.
+                // We'll return the old StringIndexCache and let the caller decide if he wants to dispose it sooner
+
+                var copy = _stringIndexCache;
                 Init();
+                return copy;
```

Comment:
Do *not* return it, in this case.
Returning it make it look like someone will dispose it, which is likely not the desirable case(someone may be using this)

Reply:
That's the idea.
There is no way to tell if someone is currently using that cache.
From the RavenDB side, I can tell if we are running a query.
https://github.com/ravendb/ravendb/pull/16646/files#diff-
3760b1669cd9af0aab370cfc92d98e3898fed9cea969a44cb19e7430aa953509R37